cardano-node/src/Cardano/CLI/Byron/UpdateProposal.hs

```diff
@@ -0,0 +1,175 @@
+
+createProtocolParametersUpdate
+  :: ProtocolParametersUpdate
+  -> [ParametersToUpdate]
+  -> ProtocolParametersUpdate
+createProtocolParametersUpdate init paramsToUpdate = go init paramsToUpdate
+ where go i [] = i
+       go i (paramToUpdate : rest) =
+         case paramToUpdate of
+           ScriptVersion val -> go i{ppuScriptVersion = Just val} rest
+           SlotDuration val -> go i{ppuSlotDuration = Just val} rest
+           MaxBlockSize val -> go i{ppuMaxBlockSize = Just val} rest
+           MaxHeaderSize val -> go i{ppuMaxHeaderSize = Just val} rest
+           MaxTxSize val -> go i{ppuMaxTxSize = Just val} rest
+           MaxProposalSize val -> go i{ppuMaxProposalSize = Just val} rest
+           MpcThd val -> go i{ppuMpcThd = Just val} rest
+           HeavyDelThd val -> go i{ppuHeavyDelThd = Just val} rest
+           UpdateVoteThd val -> go i{ppuUpdateVoteThd = Just val} rest
+           UpdateProposalThd val -> go i{ppuUpdateProposalThd = Just val} rest
+           UpdateProposalTTL val -> go i{ppuUpdateProposalTTL = Just val} rest
+           SoftforkRuleParam val -> go i{ppuSoftforkRule = Just val} rest
+           TxFeePolicy val -> go i{ppuTxFeePolicy = Just val} rest
+           UnlockStakeEpoch val -> go i{ppuUnlockStakeEpoch = Just val} rest
+
+convertProposalToGenTx :: AProposal ByteString -> Mempool.GenTx ByronBlock
+convertProposalToGenTx prop = Mempool.ByronUpdateProposal (Update.recoverUpId prop) prop
+
+createUpdateProposal
+  :: DbFile
+  -> ConfigYamlFilePath
+  -> SigningKey
+  -> [ParametersToUpdate]
+  -> ExceptT CliError IO Proposal
+createUpdateProposal dbFile configFile sKey paramsToUpdate = do
+
+  latestBlockBS <- getMostRecentBlock dbFile
+
+  latestBlock <- case decodeCBOR latestBlockBS (fromCBORABlockOrBoundary $ EpochSlots 21600) of
+                   Right (_, blk) -> right blk
+                   Left err -> left err
+
+  nc <- liftIO $ parseNodeConfigurationFP configFile
+  (genData, _) <- readGenesis $ ncGenesisFile nc
+
+  let header = abobHdrFromBlock latestBlock
+      metaData :: M.Map Update.SystemTag Update.InstallerHash
+      metaData = M.empty
+      noPassSigningKey = noPassSafeSigner sKey
+      pmId = gdProtocolMagicId genData
+      protocolParamsUpdate = createProtocolParametersUpdate emptyProtocolParametersUpdate paramsToUpdate
+
+  protocolVersion' <- hoistEither $ getProtocolVersion header
+  softwareVersion' <- hoistEither $ getSoftwareVersion header
+
+  let proposalBody = ProposalBody protocolVersion' protocolParamsUpdate softwareVersion' metaData
+
+  let proposal = signProposal pmId proposalBody noPassSigningKey
+
+  pure proposal
+
+
+getFileModificationTime :: FilePath -> IO (FilePath, UTCTime)
+getFileModificationTime fp = getModificationTime fp >>= \timeModded -> pure (fp, timeModded)
+
+-- | Gets the most recent block in the volatile database.
+getMostRecentBlock :: DbFile -> ExceptT CliError IO LByteString
+getMostRecentBlock (DbFile dbFp) = do
```

Comment:
I think on reflection we should not do it this way. We only need this to get the proposed protocol and software versions right? They can just be extra CLI args.

Reply:
Oh, I had not seen this code. It looks like this code assumes the files in the `volatile` folder correspond to blocks? That's not true, each file contains multiple blocks.

In any case, one should not nose around in that folder without going through the ChainDB, this is looking for trouble (making incorrect assumptions that are bound to get out of date).

Reply:
does this mean dropping the `--database-path` parameter entirely?

Reply:
Making the changes in https://github.com/input-output-hk/cardano-node/pull/714